thentos-core/src/Thentos/Types/Opaque.hs

```diff
@@ -0,0 +1,24 @@
+-- | Types whose internal representation is not accessible.
```

Comment:
:+1:

everybody should have one.  it's just nobody has had the patience to write them.  (and at some point in the past i argued that things were still moving too much for it to be practical.  not the case any more!)